builder/azure/arm/step_get_source_image_name.go

```diff
@@ -51,9 +63,16 @@ func (s *StepGetSourceImageName) Run(ctx context.Context, state multistep.StateB
 		if image.GalleryImageVersionProperties != nil && image.GalleryImageVersionProperties.StorageProfile != nil &&
 			image.GalleryImageVersionProperties.StorageProfile.Source != nil && image.GalleryImageVersionProperties.StorageProfile.Source.ID != nil {
 
-			imageID := *image.GalleryImageVersionProperties.StorageProfile.Source.ID
-			s.say(fmt.Sprintf(" -> SourceImageName: '%s'", imageID))
-			s.GeneratedData.Put("SourceImageName", imageID)
+			sourceID := *image.GalleryImageVersionProperties.StorageProfile.Source.ID
+			isSIGSourcedFromManagedImage, _ := regexp.MatchString("/subscriptions/[^/]*/resourceGroups/[^/]*/providers/Microsoft.Compute/images/[^/]*$", sourceID)
+			// If the Source SIG Image Version does not have its StorageProfile.Source set to a Managed Image, that means the image was directly sourced from a VM
+			// Use the SIG ID itself if there isn't a managed image that the SIG was created from
+			if !isSIGSourcedFromManagedImage {
+				sourceID = *image.ID
+			}
+
+			s.say(fmt.Sprintf(" -> SourceImageName: '%s'", sourceID))
+			s.GeneratedData.Put("SourceImageName", sourceID)
```

Comment:
What is Image.ID here? How is that different from sourceID?

I think variables can be changed here to provide a little more context, as it was a little hard for me to understand. 
```go
			sigResourceID := *image.ID
			sourceImageResourceID := *image.GalleryImageVersionProperties.StorageProfile.Source.ID
			isSIGSourcedFromManagedImage, _ := regexp.MatchString("/subscriptions/[^/]*/resourceGroups/[^/]*/providers/Microsoft.Compute/images/[^/]*$", sourceID)
			// If the Source SIG Image Version does not have its StorageProfile.Source set to a Managed Image, that means the image was directly sourced from a VM
			// Use the SIG ID itself if there isn't a managed image that the SIG was created from
			if !isSIGSourcedFromManagedImage {
				sourceImageResourceID = sigResourceID
			}

			s.say(fmt.Sprintf(" -> SourceImageName: '%s'", sourceImageResourceID))
			s.GeneratedData.Put("SourceImageName", sourceImageResourceID)
```